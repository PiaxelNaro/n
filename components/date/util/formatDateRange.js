// @flow

export default (startDate: moment$Moment, endDate: moment$Moment) => {
    if (!startDate.isSame(endDate, 'year')) {
        return `${startDate.format('MMM D, YYYY')}-${endDate.format('MMM D, YYYY')}`;
    }

    if (!startDate.isSame(endDate, 'month')) {
        return `${startDate.format('MMM D')}-${endDate.format('MMM D, YYYY')}`;
    }

    return `${startDate.format('MMM D')}-${endDate.format('D, YYYY')}`;
};
