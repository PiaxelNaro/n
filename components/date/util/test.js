// @flow
import moment from 'moment';
import formatDateRange from './formatDateRange';

describe('formatDateRange()', () => {
    it('formats date range spawning different years', () => {
        expect(
            formatDateRange(moment([2017, 2, 1]), moment([2015, 5, 7])),
        ).toMatchSnapshot();
    });

    it('formats date range spawning different months', () => {
        expect(
            formatDateRange(moment([2017, 1, 3]), moment([2017, 5, 16])),
        ).toMatchSnapshot();
    });

    it('formats date range within the same month', () => {
        expect(
            formatDateRange(moment([2017, 8, 11]), moment([2017, 8, 22])),
        ).toMatchSnapshot();
    });
});
