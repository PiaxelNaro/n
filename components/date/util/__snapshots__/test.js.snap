// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatDateRange() formats date range spawning different months 1`] = `"Feb 3-Jun 16, 2017"`;

exports[`formatDateRange() formats date range spawning different years 1`] = `"Mar 1, 2017-Jun 7, 2015"`;

exports[`formatDateRange() formats date range within the same month 1`] = `"Sep 11-22, 2017"`;
